pkg/reconciler/pipelinerun/affinity_assistant_test.go

```diff
@@ -88,6 +93,130 @@ func TestCreateAndDeleteOfAffinityAssistant(t *testing.T) {
 	}
 }
 
+// TestCreateAffinityAssistantWhenNodeIsCordoned tests an existing Affinity Assistant can identify the node failure and
+// can migrate the affinity assistant pod to a healthy node so that the existing pipelineRun runs to competition
+func TestCreateOrUpdateAffinityAssistantWhenNodeIsCordoned(t *testing.T) {
+	expectedAffinityAssistantName := getAffinityAssistantName(workspaceName, testPipelineRun.Name)
+
+	aa := []*v1.StatefulSet{{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   expectedAffinityAssistantName,
+			Labels: getStatefulSetLabels(testPipelineRun, expectedAffinityAssistantName),
+		},
+		Status: v1.StatefulSetStatus{
+			ReadyReplicas: 1,
+		},
+	}}
+
+	nodes := []*corev1.Node{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "soon-to-be-cordoned-node",
+		},
+		Spec: corev1.NodeSpec{
+			Unschedulable: true,
+		},
+	}}
+
+	p := []*corev1.Pod{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: expectedAffinityAssistantName + "-0",
+		},
+		Spec: corev1.PodSpec{
+			NodeName: "soon-to-be-cordoned-node",
+		},
+	}}
+
+	tests := []struct {
+		name, expectedError, verb, resource string
+		data                                Data
+		validatePodDeletion                 bool
+	}{{
+		name: "createOrUpdateAffinityAssistants must ignore missing affinity assistant pod, this could be interim and must not fail the entire pipelineRun",
+		data: Data{
+			StatefulSets: aa,
+			Nodes:        nodes,
+		},
+	}, {
+		name: "createOrUpdateAffinityAssistants must delete an affinity assistant pod since the node on which its scheduled is marked as unschedulable",
```

Comment:
I think any of the get/list errors here are potentially transient, and ideally shouldn't fail the entire pipelinerun. This can be for a later PR, but IMO what would make the most sense is for createOrUpdateAffinityAssistant to always return an error if it runs into an error condition, but for those errors to not always result in a permanent error in pipelinerun.go.

Reply:
thanks @lbernick yup I agree here, it will be better if pipelinerun.go can handle transient error instead of handling it this way in the `createOrUpdateAffinityAssistant`.  Since its not blocking, I will address it in a follow up PR given that I would want to introduce an additional testing in `pipelinerun_test.go` for the same.